### ISSUE TITLE
Remove CLOSED processes from task list (Reyes Camacho) (Deploying to PROD)

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#9617]
> What application are you using?

AMANDA

> Describe the problem.

Remove CLOSED processes from task list(Reyes Camacho)

Internet Browser: Edge

> Requested By
Jaclyn L.

[Attachment](https://api.knack.com/v1/applications/595d00ebd315cc4cb98daff4/download/asset/62bdbb629c5d50001f307e41/capturetl.jpg) (1.29mb)

Request ID: DTS22-104500

**Comment by Nadin-Nader** [comment]
This is something that may have to be done after the V7 upgrade. I will follow up with Jaclyn to see if this is urgent or if it can wait

**Comment by Nadin-Nader** [comment]
Checked with Jaclyn and this is still needed

**Comment by Nadin-Nader** [comment]
Asked Jaclyn. Waiting on Response

**Comment by Nadin-Nader** [comment]
@SubhaShroff  Jaclyn"s Response:

Yes that should be fine.

I have not tested out all the attempts for Case Management Review Process. I use Reject and Approval at XX% and very rarely Withdrawn. All those should technically close off the processes for reviewers so they are not on the task list.

Does that answer those questions?

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0259381 request is raised for DEV deployment.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  This is deployed in DEV and I tested for RSN: 12532418. 
When I added REJECTED attempt to case manager review process, the review processes along with the case manager review process disappeared from TaskList. It is working as expected. Please verify and let me know if I can promote it to TEST. Thanks.

**Comment by Nadin-Nader** [comment]
@SubhaShroff go ahead and send this to TEST

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0260144 is raised for TEST deployment. Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader This is deployed in TEST and tested.
RSN- 12858486 - Rejected permit case Manager process - that removed it from my tasklist.
RSN - 12874453 - Rejected Permit Case Manager process - that removed other review processes assigned to my login, it from my tasklist.

**Comment by Nadin-Nader** [comment]
According to Jaclyn the Closed TASKLIST items are still there. I let Subha know.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  As I mentioned in Teams to you, Already closed processes can be removed from Tasklist only by adding Process ENDED date, manually. The fix we did will make sure no future closed review processes or Case Manager review process will stay on TaskList. Thanks.

**Comment by Nadin-Nader** [comment]
@SubhaShroff lets send this to PROD and create a new card to run a script that puts the ENDED Date once deployed to PROD

**Comment by SubhaShroff** [comment]
Thanks @Nadin-Nader. SCTASK0261552 is raised for PROD promotion.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Following is the list of all UC processes which do not have Ended date and must be showing in the tasklist of the each user.  There are some 2400 records altogether. Is it possible to check with the user if the list is correct and that should be removed from tasklist? Then we can update all these processes ended date(which date it should be?) by executing a update script.
[Closed UC Folder Processes.csv](https://github.com/cityofaustin/atd-data-tech/files/9974169/Closed.UC.Folder.Processes.csv)

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0261552 is merged in PROD. That will serve  UC review processes/case manager review that closed from today do not display on Tasklist.
I am working on the script and soon will raise request to run it on PROD, once I am done testing it enough on Local.
Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0263527 is created and update script is sent to Yuan/Jenice to review and test it on Sandbox for one user 'CAMACHOR' i.e., (Reyes Camacho). Once this is done successfully for CAMACHOR on PROD, we can update for other users in the next request.
Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  [SCTASK0263622](https://atx.servicenowservices.com/sc_task.do?sys_id=d5fc285a87c71590287cb9d8cebb355c&sysparm_record_target=sc_task&sysparm_record_row=1&sysparm_record_rows=1&sysparm_record_list=request_item%3D89fc285a87c71590287cb9d8cebb3520%5EORDERBYnumber) is created to test this script on DEV for CAMACHOR. Thanks.

**Comment by Nadin-Nader** [comment]
Still waiting on Deployment

**Comment by TracyLinder** [comment]
This was promoted to Dev.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  I verified the Tasklist data for CAMACHOR after DEV execution and looks good to me. @Nadin-Nader  We can have a call to review before I request for TEST execution. Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0266148 is raised for TEST execution of the script. Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  TEST executed the script for CAMACHOR user. It looks all good to me on TEST. Can we verify once together and user before I submit PROD request? Thanks.

**Comment by Nadin-Nader** [comment]
Jaclyn will be out of office til the 29th of Nov. I asked Reyes, if he doesnt respond i will reach out to Darren

**Comment by Nadin-Nader** [comment]
Asked Jaclyn. Waiting on Response

**Comment by Nadin-Nader** [comment]
@SubhaShroff i spoke to Jaclyn and she said it looks good and to send the changes/run scripts to/on PROD

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0268109 is submitted for PROD execution - CAMACHOR user.

**Comment by SubhaShroff** [comment]
@Nadin-Nader for Reyes user the script is executed on PROD. If the user can check it we can go ahead for other users. Thanks.

**Comment by Nadin-Nader** [comment]
Sent message to Jaclyn to have Reyes or her test this before we run the script for everyone else. Waiting on Response

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0268761 is submitted for SANDBOX execution of the script for all the users.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0269257 is submitted for DEV execution of the script for all users. Before that I have verified the above deployment in SANDBOX for all the users, listed.

**Comment by SubhaShroff** [comment]
Status:
Waiting on User's response to run the script on PROD.

**Comment by Nadin-Nader** [comment]
@SubhaShroff Jaclyn got back to me and said the following:

> [1:29 PM] Lozano, Jaclyn
> I think it's just these that need to be removed. 
> ![MicrosoftTeams-image (34).png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/ce291ba8-d687-44a8-82d2-c0d08045c6e7)
> 
> [1:32 PM] Lozano, Jaclyn
> When the folder status is active, the process is complete for Utility Coordination. 
> 
> [1:33 PM] Lozano, Jaclyn
> He should only have tasks for folder status = IN REVIEW

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The above shown(highlighted) processes look like closed after the script is run. I am not sure how the user is closing these processes, I don't see any attempt added. But, its only changing the status of the process not updating the
The process should be closed by adding attempt - that will add end date to the process. That will take off the record from Task list. 
I remember once Jaclyn showing us that they can close the processes from Tasklist. I think that is not helping.  Will talk to you about it in detail in the standup tomorrow. 
Thank you.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  After the call with Jaclyn, I added the change in ISSUE procedure. SCTASK0281135 is submitted DEV/TEST Deployment of the commit - [ac5c5027dba2](http://coaamandadevops02.austintexas.gov/R5:ac5c5027dba22e35bd08058dc9320ab0edba35c1).

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been deployed to DEV/TEST, please UAT

**Comment by Nadin-Nader** [comment]
Approved to run Script.
Next Steps:

- Run script on DEV/TEST
- Then; Run script on PROD

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0283069 is submitted for DEV - script execution for all users.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The script is executed for all the users on DEV. I randomly, tested for the following users and all 'closed' UC processes are removed from the tasklist. Since you already approved I am sending this script to run in TEST.
Richard Hengst
Daren Duncan
Richard Hengst
Julian Howser
Isaiah Lewallen
Tim McGowan

**Comment by SubhaShroff** [comment]
@Nadin-Nader The script is executed for all the users on TEST. I tested for the following users and all 'closed' UC processes are removed from the tasklist. This is ready for UAT, for all the users.
Richard Hengst
Daren Duncan
Richard Hengst
Julian Howser
Isaiah Lewallen
Tim McGowan

**Comment by Nadin-Nader** [comment]
I et Jaclyn know that the script was ran on TEST and on who. I asked her to verify and approve for PROD. Waiting on response

**Comment by Nadin-Nader** [comment]
@SubhaShroff Spoke to Jaclyn and she said this was good to go

**Comment by SubhaShroff** [comment]
@Nadin-Nader  [SCTASK0288688](https://atx.servicenowservices.com/sc_task.do?sys_id=9b4da6d78701a510287cb9d8cebb3599&sysparm_record_target=sc_task&sysparm_record_row=1&sysparm_record_rows=1&sysparm_record_list=request_item%3D8f4da6d78701a510287cb9d8cebb3520%5EORDERBYnumber) Is requested for PROD script execution.
SCTASK0288690 - is requested to PROD deploy the commit done for ISSUE_psm changes.

**Comment by Nadin-Nader** [comment]
Update:
Still waiting on CTM to deploy to PROD

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  This ticket related script is executed for all the listed users on PROD and Issue procedure is also updated to take care that no closed review processes display on the tasklists. Now users can verify. Thanks.

**Comment by Nadin-Nader** [comment]
Informed Jaclyn that this was on PROD and that we will be closing it.
